src/sql/workbench/services/notebook/browser/models/notebookModel.ts

```diff
@@ -538,8 +540,119 @@ export class NotebookModel extends Disposable implements INotebookModel {
+
+				let model = editorControl.getModel() as TextModel;
+				let range = model.getFullModelRange();
+				let selection = editorControl.getSelection();
+				let source = this.cells[index].source;
+				let newcell = null, tailcell = null, partial = null;
+				let newcellindex = index;
+				let tailcellindex = index;
+
+				// Save UI state
+				let showMarkdown = this.cells[index].showMarkdown;
+				let showPreview = this.cells[index].showPreview;
+
+				//Get selection value from current cell
+				let newcellcontent = model.getValueInRange(selection);
+
+				//Get content after selection
+				let tailrange = range.setStartPosition(selection.endLineNumber, selection.endColumn);
+				let tailcellcontent = model.getValueInRange(tailrange);
+
+				//Get content before selection
+				let headrange = range.setEndPosition(selection.startLineNumber, selection.startColumn);
+				let headcontent = model.getValueInRange(headrange);
+
+				//Set content before selection if the selection is not the same as original content
+				if (headcontent.length) {
+					let headsource = source.slice(range.startLineNumber - 1, selection.startLineNumber - 1);
+					if (selection.startColumn > 1) {
+						partial = source.slice(selection.startLineNumber - 1, selection.startLineNumber)[0].slice(0, selection.startColumn - 1);
+						headsource = headsource.concat(partial.toString());
+					}
+					this.cells[index].source = headsource;
+				}
+
+				if (newcellcontent.length) {
+					newcell = this.createCell(cellType);
+					let newsource = source.slice(selection.startLineNumber - 1, selection.endLineNumber) as string[];
+					if (selection.startColumn > 1) {
+						partial = source.slice(selection.startLineNumber - 1)[0].slice(selection.startColumn - 1);
+						newsource.splice(0, 1, partial);
+					}
+					if (selection.endColumn !== source[selection.endLineNumber - 1].length) {
+						let splicestart = 0;
+						if (selection.startLineNumber === selection.endLineNumber) {
+							splicestart = selection.startColumn - 1;
+						}
+						let partial = source.slice(selection.endLineNumber - 1, selection.endLineNumber)[0].slice(splicestart, selection.endColumn - 1);
+						newsource.splice(newsource.length - 1, 1, partial);
+					}
+					newcell.source = newsource;
+					newcellindex++;
+					this.insertCell(newcell, newcellindex);
+				}
+
+				if (tailcellcontent.length) {
+					//tail cell will be of original cell type.
+					tailcell = this.createCell(this._cells[index].cellType);
+					let tailsource = source.slice(tailrange.startLineNumber - 1) as string[];
+					if (selection.endColumn > 1) {
+						partial = source.slice(tailrange.startLineNumber - 1, tailrange.startLineNumber)[0].slice(tailrange.startColumn - 1);
+						tailsource.splice(0, 1, partial);
+					}
+					tailcell.source = tailsource;
+					tailcellindex = newcellindex + 1;
+					this.insertCell(tailcell, tailcellindex);
+				}
+
+				//Delete the original cell if the selection begins at the start
```

Comment:
Why not just leave the selected text in the original cell in this case and only create a new cell for the rest of the text?

Reply:
There is some edge condition where this happens. Hence this code.

Reply:
...what edge condition? Please make sure to document stuff like that since it's not obvious just from reading the code.

Reply:
Ping, I'm still unsure why this is necessary. It adds more complexity so if we can remove it that would be best.

Reply:
The above code is required when user selects the text from the beginning of the cell. The logic empties original cell which is removed here.

Reply:
@chlafreniere Are you ok with this?

Reply:
@rajeshka my concern is really maintainability here since my team will still own this code, and it makes me a bit nervous since we don't have unit tests yet. When do you plan on doing WYSIWYG cell splitting?